Sources/BraveWallet/Crypto/Stores/TransactionConfirmationStore.swift

```diff
@@ -46,10 +46,14 @@ public class TransactionConfirmationStore: ObservableObject {
   /// This is an id for the unppproved transaction that is currently displayed on screen
   @Published var activeTransactionId: BraveWallet.TransactionInfo.ID = "" {
     didSet {
-      if let tx = transactions.first(where: { $0.id == activeTransactionId }) {
+      if let tx = allTxs.first(where: { $0.id == activeTransactionId }) {
         updateTransaction(with: tx)
-      } else if let firstTx = transactions.first {
+        activeTxStatus = tx.txStatus
+      } else if let firstTx = unapprovedTxs.first {
         updateTransaction(with: firstTx)
+        activeTxStatus = firstTx.txStatus
+      } else {
+        activeTxStatus = .error
```

Comment:
Is this displayed in the UI? Concern is that it would show as Tx Confirmation is dismissed when transactions count is 0, which would fall into this case

Reply:
added failed state (haven't heard back from james with his feedback with the design changes we came up with. but will address next week if there is any) [29dc327](https://github.com/brave/brave-ios/pull/6960/commits/29dc3270f2191a8677bd908927acd01b0fc1f7de)

Reply:
The error state gets shown during dismiss of `eth_signTransaction`
![Simulator Screen Shot - iPad Pro (11-inch) (3rd generation) - 2023-02-20 at 17 10 43](https://user-images.githubusercontent.com/5314553/220373120-df1c20dd-c84e-4c96-932a-4238a9fe14bd.png)

Reply:
The error state gets shown during dismiss of `eth_signTransaction` (slow animations toggle used to show it easier)

https://user-images.githubusercontent.com/5314553/220373610-1814915a-5857-456f-a334-b346d3ce088d.mp4

Reply:
fixed in [cf0e2bb](https://github.com/brave/brave-ios/pull/6960/commits/cf0e2bb683b9aac63b9ea363fdca4e7c43eb009b)